docs/reference/settings/common-defs.asciidoc

```diff
@@ -0,0 +1,167 @@
+
+tag::ssl-keystore-path[]
+The path for the keystore file that contains a private key and certificate.
+//TBD: It must be either a Java keystore (jks) or a PKCS#12 file.
+//TBD: You cannot use this setting and `ssl.key` at the same time.
+end::ssl-keystore-path[]
+
+tag::ssl-keystore-secure-key-password[]
+The password for the key in the keystore. The default is the keystore password.
+//TBD: You cannot use this setting and `ssl.keystore.key_password` at the same time.
+end::ssl-keystore-secure-key-password[]
+
+tag::ssl-keystore-secure-password[]
+The password for the keystore.
+//TBD: You cannot use this setting and `ssl.keystore.password` at the same time.
+end::ssl-keystore-secure-password[]
+
+tag::ssl-keystore-type-pkcs11[]
+The format of the keystore file. For the Java keystore format, use `jks`. For
+PKCS#12 files, use `PKCS12`. For a PKCS#11 token, use `PKCS11`. The default is
+`jks`.
+end::ssl-keystore-type-pkcs11[]
+
+tag::ssl-keystore-type-pkcs12[]
+The format of the keystore file. It must be either `jks` or `PKCS12`. If the
+keystore path ends in ".p12", ".pfx", or "pkcs12", this setting defaults 
+to `PKCS12`. Otherwise, it defaults to `jks`.
+end::ssl-keystore-type-pkcs12[]
+
+tag::ssl-secure-key-passphrase[]
+The passphrase that is used to decrypt the private key. Since the key might not
+be encrypted, this value is optional. 
+//TBD: You cannot use this setting and `ssl.key_passphrase` at the same time.
+end::ssl-secure-key-passphrase[]
+
+tag::ssl-supported-protocols[]
+Supported protocols with versions. Valid protocols: `SSLv2Hello`,
+`SSLv3`, `TLSv1`, `TLSv1.1`, `TLSv1.2`, `TLSv1.3`. If the JVM supports TLSv1.3,
+the default is `TLSv1.3,TLSv1.2,TLSv1.1`. Otherwise, the default is
+`TLSv1.2,TLSv1.1`.
++
+--
+NOTE: If `xpack.security.fips_mode.enabled` is `true`, you cannot use `SSLv2Hello` 
+or `SSLv3`. See <<fips-140-compliance>>.
+
+--
+end::ssl-supported-protocols[]
+//TBD: Are these three different definitions for supported protocols valid?
+tag::ssl-supported-protocols-1[]
+Supported protocols for TLS/SSL (with versions). Defaults to `TLSv1.3,TLSv1.2,TLSv1.1`.
+end::ssl-supported-protocols-1[]
+
+tag::ssl-supported-protocols-2[]
+Specifies the supported protocols for TLS/SSL. If the JVM supports TLSv1.3, the 
+defaults is `TLSv1.3,TLSv1.2,TLSv1.1`. Otherwise, the default is
+`TLSv1.2,TLSv1.1`.
+end::ssl-supported-protocols-2[]
+
+tag::ssl-truststore-password[]
+The password for the truststore.
+//TBD: You cannot use this setting and `ssl.truststore.secure_password` at the same time.
+end::ssl-truststore-password[]
+
+tag::ssl-truststore-path[]
+The path for the keystore that contains the certificates to trust. It must be
+either a Java keystore (jks) or a PKCS#12 file.
+//TBD: You cannot use this setting and `ssl.certificate_authorities` at the same time.
+end::ssl-truststore-path[]
+
+tag::ssl-truststore-path-jks[]
+The path for the Java keystore file that contains the certificates to trust.
```

Comment:
I don't think we need this. If it's used somewhere it's probably out of date.

Reply:
It's used in the ssl.truststore.path in http://elasticsearch_51017.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-settings.html and the {ssl-prefix-ssl}.truststore.path in http://elasticsearch_51017.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/notification-settings.html and http://elasticsearch_51017.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/monitoring-settings.html

I'll switch those occurrence to using the "ssl-truststore-path" shared definition instead.